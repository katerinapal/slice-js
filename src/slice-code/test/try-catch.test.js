import {runAllCombosTests, comboOfBools} from './helpers/utils'

// fit(
//   'tryCatch(false)',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/try-catch'), ({tryCatch}) => {
//     return [tryCatch(false)]
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/try-catch'),
  methods: [
    {methodName: 'tryCatch', possibleArguments: comboOfBools(1)},
    {methodName: 'statementsAfterThrow', explicitArgs: [[]]},
  ],
})
