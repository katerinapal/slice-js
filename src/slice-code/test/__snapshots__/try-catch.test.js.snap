// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statementsAfterThrow() 1`] = `
"export {statementsAfterThrow}

function statementsAfterThrow() {
  const ret = {}
  try {
    ret.before = true

    throw new Error('throw error')
  } catch (error) {
    ret.error = error
  } finally {
    return ret
  }
}
"
`;

exports[`tryCatch(false) && tryCatch(true) 1`] = `
"export {tryCatch}

function tryCatch(shouldThrow) {
  const ret = {}
  try {
    if (shouldThrow) {
      throw new Error('throw error')
    }
    ret.noThrow = true
  } catch (error) {
    ret.error = error
    return error
  } finally {
    return ret
  }
}
"
`;

exports[`tryCatch(false) 1`] = `
"export {tryCatch}

function tryCatch() {
  const ret = {}

  ret.noThrow = true

  return ret
}
"
`;

exports[`tryCatch(true) 1`] = `
"export {tryCatch}

function tryCatch() {
  const ret = {}
  try {
    throw new Error('throw error')
  } catch (error) {
    ret.error = error
    return error
  } finally {
    return ret
  }
}
"
`;
